test/splitgraph/commands/test_aggregation_pushdown.py

```diff
@@ -0,0 +1,392 @@
+            "avg.balance": {"avg": {"field": "balance"}},
+            "max.balance": {"max": {"field": "balance"}},
+            "sum.balance": {"sum": {"field": "balance"}},
+            "min.age": {"min": {"field": "age"}},
+            "avg.age": {"avg": {"field": "age"}},
+        }
+    }
+
+    # Ensure results are correct
+    result = get_engine().run_sql(query)
+    assert len(result) == 1
+
+    # Assert aggregation result
+    assert result[0] == (999, 25714.837, 49989, 25714837, 20, Decimal("30.171"))
+
+
+@pytest.mark.mounting
+def test_elasticsearch_gropuing_clauses_only(local_engine_empty):
+    _mount_elasticsearch()
+
+    # Single column grouping
+    query = "SELECT state FROM es.account GROUP BY state"
+
+    # Ensure grouping is going to be pushed down
+    result = get_engine().run_sql("EXPLAIN " + query)
+    assert len(result) > 2
+    assert _extract_query_from_explain(result) == {
+        "aggs": {
+            "group_buckets": {
+                "composite": {"sources": [{"state": {"terms": {"field": "state"}}}], "size": 5}
+            }
+        }
+    }
+
+    # Ensure results are correct
+    result = get_engine().run_sql(query, return_shape=ResultShape.MANY_ONE)
+    assert len(result) == 51
+
+    # Assert aggregation result
+    assert result == [
+        "AK",
+        "AL",
+        "AR",
+        "AZ",
+        "CA",
+        "CO",
+        "CT",
+        "DC",
+        "DE",
+        "FL",
+        "GA",
+        "HI",
+        "IA",
+        "ID",
+        "IL",
+        "IN",
+        "KS",
+        "KY",
+        "LA",
+        "MA",
+        "MD",
+        "ME",
+        "MI",
+        "MN",
+        "MO",
+        "MS",
+        "MT",
+        "NC",
+        "ND",
+        "NE",
+        "NH",
+        "NJ",
+        "NM",
+        "NV",
+        "NY",
+        "OH",
+        "OK",
+        "OR",
+        "PA",
+        "RI",
+        "SC",
+        "SD",
+        "TN",
+        "TX",
+        "UT",
+        "VA",
+        "VT",
+        "WA",
+        "WI",
+        "WV",
+        "WY",
+    ]
+
+    # Multi-column grouping
+    query = "SELECT gender, age FROM es.account GROUP BY age, gender"
+
+    # Ensure grouping is going to be pushed down
+    result = get_engine().run_sql("EXPLAIN " + query)
+    assert len(result) > 2
+    assert _extract_query_from_explain(result) == {
+        "aggs": {
+            "group_buckets": {
+                "composite": {
+                    "sources": [
+                        {"gender": {"terms": {"field": "gender"}}},
+                        {"age": {"terms": {"field": "age"}}},
+                    ],
+                    "size": 5,
+                }
+            }
+        }
+    }
+
+    # Ensure results are correct
+    result = get_engine().run_sql(query)
+    assert len(result) == 42
+
+    # Assert aggregation result
+    gender = "F"
+    age = 20
+    for row in result:
+        assert row == (gender, age)
+
+        age += 1
```

Comment:
This assertion is kind of difficult to mentally unroll. I recently started using `pytest-snapshot` (https://pypi.org/project/pytest-snapshot/, usage example: https://github.com/splitgraph/splitgraph/blob/master/test/splitgraph/cloud/project/test_merging.py#L21) that can store expected outputs as files and then easily regenerate them, so that you can essentially approve a change to a test through version control and see what changed. I think it's worth changing some of these (especially the bigger query outputs where you're asserting the whole output) to use pytest-snapshot.

Reply:
Fair point, checking out `pytest-snapshot`